src/omv/imlib/draw.c

```diff
@@ -3140,6 +3140,15 @@ void imlib_draw_image(image_t *dst_img,
         void *data = fb_alloc_all(&size, FB_ALLOC_PREFER_SPEED | FB_ALLOC_CACHE_ALIGN);
         fb_free();
 
+        // Leave RAM reserved for the GPU.
+        #if (OMV_GPU_ENABLE == 1 && OMV_GPU_RESERVED_FB_ALLOC_BYTES > 0)
+        if (size > OMV_GPU_RESERVED_FB_ALLOC_BYTES) {
+            size -= OMV_GPU_RESERVED_FB_ALLOC_BYTES;
+        } else {
+            fb_alloc_fail();
+        }
+        #endif
+
         // Allocate a buffer to hold chunks of the transposed image while not using all of the on-chip RAM.
         size = IM_MIN(size, image_size(&t_src_img));
```

Comment:
This limits the RAM usage during transpose to what is actually needed. The more RAM transpose has to work with up to the image size the faster it goes. However, if there's less RAM available than the image size then this uses up all fb_alloc space available still so it's not enough to ensure the GPU has RAM for CLUT loading or etc.

Reply:
It's only 1KB if it was reserved in the DMA memory area that is algined. But, DMA RAM is somewhat tight on some of our models. Might be hard to fit 1KB on the older boards. I think this fix is better than reserving RAM.

Reply:
Huh, it all fits in DMA memory. Okay, guess this is settled. Just need to retest.

Reply:
Yes, that's what it's for. Note OMV3 uses DTCM for this, so you may want to fix it.

Reply:
I get an instant reset on the Giga with this... debugging.

Reply:
OMV3 works with this since there's an AHB_S port on the CPU which allows DMA access to the DTCM.

Test the OMV3 and H7P and they seem to be fine.

Reply:
Resolved.